### ISSUE TITLE
[Feature Request] When moving stuff around in IDEA automatically modify paths inside tscn files

### ISSUE THREAD

**Issue by Humberd** [utopia-rise/godot-kotlin-jvm#204]
It would be cool for Godot to not complain abound incorrect resource paths when moving a path or a scene file around

**Comment by CedNaru** [comment]
That would be nice to have, but I don't think it's a feature related to Kotlin itself.
It's up to Godot to improve their server language so we have a better way to interact with Godot from an external editor.

**Comment by Humberd** [comment]
What I mean is that `.tscn` files, which include paths to kotlin files like this

```tscn
[ext_resource path="res://src/foo/bar/Cat.kt" type="Texture" id=2]
[ext_resource path="res://src/main/kotlin/clientjvm/scenes/GroundTileScene.kt" type="Script" id=3]
```

It would be nice for the plugin to automatically modify them

**Comment by CedNaru** [comment]
I don't think it's going to work easily. If you have the scene opened in Godot at the same time, I'm pretty sure Godot won't take that external change into account and basically overwrite it the next time the Godot editor saves.

**Comment by Humberd** [comment]
Damn it. Thanks for clarification

**Comment by chippmann** [comment]
Actually we already have that in the IDE plugin. So it should "work". You could try it by leaving the godot editor closed, move a class which is referenced in a scene file (has to have `tscn` extension) and see if it changes.
But as @CedNaru said: Godot in its current state is notoriously bad at picking up external file changes. Most of the time your changes either get ignored or outright overridden.

That said if the above mentioned refactoring does not work when the editor is closed, its a bug. But if it does, there is nothing we can do until its fixed in godot itself.
Could you try it?

**Comment by chippmann** [comment]
![1](https://user-images.githubusercontent.com/22662033/116354778-32729f00-a7f9-11eb-8d58-c89f02b6cdd4.png)
Works for me. But could you try as well?

**Comment by raniejade** [comment]
Is there a ticket already filed in godot that we can track? I don't we can support it Godot just ignores external changes.

**Comment by piiertho** [comment]
That's pretty much the same problem on inheritance ticket: https://github.com/utopia-rise/godot-kotlin-jvm/issues/198
An issue to Godot might be a good idea, but I dont think it will be implemented fast

**Comment by chippmann** [comment]
> That's pretty much the same problem on inheritance ticket: #198
> An issue to Godot might be a good idea, but I dont think it will be implemented fast

I don't agree. Technically it's a similar problem, but function wise i don't want to support #192 as stated on discord and hence in the comment there.

**Comment by piiertho** [comment]
I see no point of no supporting #198 if we support this one. To me that's a Godot issue, so Godot should provide better interfacing with external editors. So we should better write an issue to Godot rather introducing gratuites which Can have side effects.

**Comment by chippmann** [comment]
> I see no point of no supporting #198 if we support this one. To me that's a Godot issue, so Godot should provide better interfacing with external editors. So we should better write an issue to Godot rather introducing feature which Can have side effects.

Again i disagree. The point for me is not the refactoring feature itself, or godot not picking up the external change. The problem i have with #198 is that we cannot just change the type of a node based on the script. That's what we've discussed on discord. Not the actual change of the scene file itself. Which is the topic of this issue.
The problem with #198 is that there is no system in godot to refactor node types based on script inheritance. In no language that godot supports. If we add this, even if godot fixes their problem with external file changes, we still introduce unwanted sideeffects. The relevant part of the discussion to which i'm reffering started [here](https://discord.com/channels/675058327088136212/675367413193375765/836238485886861402).

So to sum up: IMO this issue and #198 are two completely separate things. The problem is not the external file change we do in the case of #198 it's a different problem. The topic of this issue on the other hand is in fact external file change. Two different things. Hence i don't want to support #198 atm. This issue on the otherhand is definitely something we can support imo.

**Comment by piiertho** [comment]
After reading the discord conversation all makes sense to me, thx !
Let's close #198

**Comment by chippmann** [comment]
Closing as the refactoring on our side works. If changes are not picked up by the godot editor while it's open, that's an issue we can't deal with from the plugin perspective.
